apps/wallpaper/js/share.js

```diff
@@ -117,6 +119,9 @@ window.onload = function() {
   }
 
   function scaleImage() {
+      // Disable set wallpaper button to prevent hammering and causing multiple
+      // calls to scaleImage.
+      setButton.disabled = true;
```

Comment:
Nice find. Thanks for fixing this.

One minor nit.  The line that creates the canvas doesn't have a canvas and looks like it is grouped with this disable code. If it was me, I'd move the createElement() call down three lines, and update the comment below so it talks about creating a canvas that is exactly the size of the screen in device pixels.